plugin-mongodb/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+use alumet::{
+    measurement::{AttributeValue, MeasurementBuffer, WrappedMeasurementValue},
+    pipeline::{
+        elements::{error::WriteError, output::OutputContext},
+        Output,
+    },
+    plugin::rust::{deserialize_config, serialize_config, AlumetPlugin},
+};
+
+use mongodb::{bson::doc, sync::Client};
+use mongodb2::convert_timestamp;
+use serde::{Deserialize, Serialize};
+
+mod mongodb2;
+
+pub struct MongoDbPlugin {
+    config: Option<Config>,
+}
+
+impl AlumetPlugin for MongoDbPlugin {
+    fn name() -> &'static str {
+        "mongodb"
+    }
+
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    fn default_config() -> anyhow::Result<Option<alumet::plugin::ConfigTable>> {
+        Ok(Some(serialize_config(Config::default())?))
+    }
+
+    fn init(config: alumet::plugin::ConfigTable) -> anyhow::Result<Box<Self>> {
+        let config = deserialize_config(config)?;
+        Ok(Box::new(MongoDbPlugin { config: Some(config) }))
+    }
+
+    fn start(&mut self, alumet: &mut alumet::plugin::AlumetPluginStart) -> anyhow::Result<()> {
+        let config = self.config.take().unwrap();
+
+        // Test connection to MongoDb.
+        let uri = mongodb2::build_mongo_uri(&config);
+        let client = Client::with_uri_str(uri)?;
+        let rt = tokio::runtime::Builder::new_current_thread().enable_all().build()?;
+        log::info!("Testing connection to MongoDb...");
+        rt.block_on(async {
+            let collection = client.database(&config.database).collection(&config.collection);
+            collection.insert_one(doc! {}).await.unwrap();
+        });
+        log::info!("Test successful.");
+
+        // Create the output.
+        alumet.add_blocking_output(Box::new(MongoDbOutput {
+            client: Client::with_uri_str(mongodb2::build_mongo_uri(&config)).unwrap(),
+            database: config.database,
+            collection: config.collection,
+        }));
+        Ok(())
+    }
+
+    fn stop(&mut self) -> anyhow::Result<()> {
+        Ok(())
+    }
+}
+
+struct MongoDbOutput {
+    client: Client,
+    database: String,
+    collection: String,
+}
+
+impl Output for MongoDbOutput {
+    fn write(&mut self, measurements: &MeasurementBuffer, ctx: &OutputContext) -> Result<(), WriteError> {
+        // Cannot write anything with an empty buffer.
+        if measurements.is_empty() {
+            log::warn!("MongoDb recieved an empty MeasurementBuffer");
+            return Ok(());
+        }
+
+        // Build the data to send to MongoDb
+        let mut docs = vec![];
+        for m in measurements {
+            let mut doc = doc! {};
```

Comment:
```suggestion
            let mut doc = Document::new();
```
personal preference (minor nitpick :D): the document is empty here, so the macro does the same as `Document::new()` but obscures the type and adds (compilation) overhead.